File: bench/fixtures/small/simple.ts
// Small TypeScript fixture (<100 LOC)
// Simple function with basic constraints

interface User {
    id: number;
    name: string;
    email: string;
}

function validateEmail(email: string): boolean {
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    return emailRegex.test(email);
}

function createUser(name: string, email: string): User | null {
    if (!validateEmail(email)) {
        return null;
    }
    
    return {
        id: Math.floor(Math.random() * 10000),
        name: name.trim(),
        email: email.toLowerCase(),
    };
}

export { User, createUser, validateEmail };

File: bench/fixtures/small/simple.py
# Small Python fixture (<100 LOC)
# Simple class with validation

import re
from typing import Optional
from dataclasses import dataclass

@dataclass
class User:
    id: int
    name: str
    email: str

def validate_email(email: str) -> bool:
    """Validate email format using regex."""
    pattern = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
    return bool(re.match(pattern, email))

def create_user(name: str, email: str) -> Optional[User]:
    """Create a new user with validation."""
    if not validate_email(email):
        return None
    
    import random
    return User(
        id=random.randint(1, 10000),
        name=name.strip(),
        email=email.lower()
    )
